code/Player/Cameras/ThirdPersonSpectateCamera.cs

```diff
@@ -2,63 +2,59 @@
 
 namespace TTT;
 
-public class ThirdPersonSpectateCamera : CameraMode, ISpectateCamera
+public partial class ThirdPersonSpectateCamera : CameraMode, ISpectateCamera
 {
-	private const int CameraDistance = 120;
+	public Player InitialSpectatedPlayer { get; set; }
+
+	private Player _owner;
 	private Vector3 _targetPos;
 	private Angles _lookAngles;
 
-	public override void Activated()
+	protected override void OnActivate()
 	{
-		base.Activated();
-
-		if ( Local.Pawn is not Player player )
-			return;
+		_owner = Entity as Player;
 
-		player.UpdateSpectatedPlayer();
+		if ( InitialSpectatedPlayer.IsValid() )
+			_owner.CurrentPlayer = InitialSpectatedPlayer;
+		else
+			_owner.UpdateSpectatedPlayer();
 	}
 
 	public override void Deactivated()
 	{
-		if ( Local.Pawn is not Player player )
-			return;
-
-		player.CurrentPlayer = null;
+		// We don't need to remove the current player if we are swapping
+		// to a FirstPersonSpectatorCamera. It should be the same player.
+		if ( _owner.Camera is not FirstPersonSpectatorCamera )
+			_owner.CurrentPlayer = null;
```

Comment:
@kolexxx I think this is the only weird thing. This occurs after we've already swapped cameras, so we need to avoid setting `CurrentPlayer` to null.